src/test/java/io/jenkins/docker/pipeline/DockerAgentTest.java

```diff
@@ -0,0 +1,50 @@
+package io.jenkins.docker.pipeline;
+
+import com.nirima.jenkins.plugins.docker.DockerCloud;
+import io.jenkins.docker.client.DockerAPI;
+import java.util.Collections;
+import org.jenkinsci.plugins.docker.commons.credentials.DockerServerEndpoint;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.JenkinsRule;
+
+public class DockerAgentTest {
+
+    @BeforeClass
+    public static void before() {
+        DockerNodeStepTest.before();
+    }
+
+    @ClassRule
+    public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    @Rule
+    public JenkinsRule r = new JenkinsRule();
+
+    @Test
+    public void smokes() throws Exception {
+        // as in DockerNodeStepTest.defaults:
+        r.jenkins.clouds.add(new DockerCloud("whatever", new DockerAPI(new DockerServerEndpoint("unix:///var/run/docker.sock", null)), Collections.emptyList()));
+        WorkflowJob j = r.createProject(WorkflowJob.class, "p");
+        j.setDefinition(new CpsFlowDefinition(
+            "pipeline {\n" +
+            "  agent {\n" +
+            "    dockerContainer 'openjdk:8'\n" +
+            "  }\n" +
+            "  stages {\n" +
+            "    stage('whatever') {\n" +
+            "      steps {\n" +
+            "        sh 'java -version'\n" +
```

Comment:
This is the basic usage. Pretty straightforward in basic cases: just replace `docker` with `dockerContainer` _and test_.